src/fsevent.rs

```diff
@@ -0,0 +1,369 @@
+#![allow(non_upper_case_globals, dead_code)]
+// via https://github.com/octplane/fsevent-rust
+
+use fsevent_sys::core_foundation as cf;
+use fsevent_sys::fsevent as fs;
+use std::slice;
+use std::ffi::CString;
+use std::mem::transmute;
+use std::slice::from_raw_parts_mut;
+use std::str::from_utf8;
+use std::ffi::CStr;
+use std::convert::AsRef;
+use std::thread;
+use std::sync::{Arc, RwLock};
+
+use std::sync::mpsc::{channel, Sender, Receiver};
+use super::{Error, Event, op, Watcher};
+use std::path::{Path, PathBuf};
+use libc;
+
+
+// TODO: add this to fsevent_sys
+const kCFStringEncodingUTF8: u32 = 0x08000100;
+pub type CFStringEncoding = u32;
+
+const kCFCompareEqualTo: i32     = 0;
+pub type CFComparisonResult = i32;
+
+// MacOS uses Case Insensitive path
+const kCFCompareCaseInsensitive: u32 = 1;
+pub type CFStringCompareFlags = u32;
+
+#[link(name = "CoreServices", kind = "framework")]
+extern "C" {
+  pub fn FSEventStreamInvalidate(streamRef: fs::FSEventStreamRef);
+  pub fn FSEventStreamRelease(streamRef: fs::FSEventStreamRef);
+  pub fn FSEventStreamUnscheduleFromRunLoop(streamRef: fs::FSEventStreamRef, runLoop: cf::CFRunLoopRef, runLoopMode: cf::CFStringRef);
+
+  pub fn CFRunLoopStop(rl: cf::CFRunLoopRef);
+  pub fn CFShowStr (str: cf::CFStringRef);
+  pub fn CFStringGetCStringPtr(theString: cf::CFStringRef, encoding: CFStringEncoding) -> *const libc::c_char;
+  pub fn CFStringCompare(theString1: cf::CFStringRef, theString2: cf::CFStringRef, compareOptions: CFStringCompareFlags) -> CFComparisonResult;
+  pub fn CFArrayRemoveValueAtIndex(theArray: cf::CFMutableArrayRef, idx: cf::CFIndex);
+}
+
+pub const NULL: cf::CFRef = cf::NULL;
+
+unsafe fn str_path_to_cfstring_ref(source: &str) -> cf::CFStringRef {
+  let c_path = CString::new(source).unwrap();
+  let c_len = libc::strlen(c_path.as_ptr());
+  let mut url = cf::CFURLCreateFromFileSystemRepresentation(cf::kCFAllocatorDefault, c_path.as_ptr(), c_len as i64, false);
+  let mut placeholder = cf::CFURLCopyAbsoluteURL(url);
+  cf::CFRelease(url);
+
+  let imaginary: cf::CFRef = cf::CFArrayCreateMutable(cf::kCFAllocatorDefault, 0, &cf::kCFTypeArrayCallBacks);
+
+  while !cf::CFURLResourceIsReachable(placeholder, cf::kCFAllocatorDefault) {
+
```

Comment:
Notify is actually all in 2 space indent, probably because that's how I started it. I should switch to the more idiomatic 4 space indent at some point, but not right now. This comment was referring to the blank line after the `while`. I fixed all these minor style issues in the release commit.